cpp/src/utilities/bit_util.cuh

```diff
@@ -37,80 +42,79 @@
 namespace gdf {
 namespace util {
 
-static constexpr int ValidSize = 32;
-using ValidType = uint32_t;
+template <typename T>
+constexpr inline std::size_t size_in_bits() { return sizeof(T) * CHAR_BIT; }
 
+template <typename T>
+constexpr inline std::size_t size_in_bits(const T&) { return size_in_bits<T>(); }
 
-// Instead of this function, use gdf_valid_allocation_size from legacy_bitmask.hpp
-//CUDA_HOST_DEVICE_CALLABLE
-//  size_t
-//  valid_size(size_t column_length)
-//{
-//  const size_t n_ints = (column_length / ValidSize) + ((column_length % ValidSize) ? 1 : 0);
-//  return n_ints * sizeof(ValidType);
-//}
+namespace detail {
 
-// Instead of this function, use gdf_is_valid from gdf/utils.h
-///CUDA_HOST_DEVICE_CALLABLE bool get_bit(const gdf_valid_type* const bits, size_t i)
-///{
-///  return  bits == nullptr? true :  bits[i >> size_t(3)] & (1 << (i & size_t(7)));
-///}
+template <typename BitContainer, typename Size>
+constexpr CUDA_HOST_DEVICE_CALLABLE
+Size intra_container_index(Size bit_index) { return bit_index % size_in_bits<BitContainer>(); }
```

Comment:
I dislike the name `intra_container_index`. I think the name should express that it's the index of a bit within a container. The function name needs "bit" in it somewhere.

Also naming the input parameter `bit_index` is a bit confusing. It's actually the element/record index from the column of elements. The thing returned is the bit index within the container.